examples/js/GPUComputationRenderer.js

```diff
@@ -160,136 +190,166 @@ function GPUComputationRenderer( sizeX, sizeY, renderer ) {
-					uniforms[ depVar.name ].value = depVar.renderTargets[ currentTextureIndex ].texture;
+		}
 
-				}
+		// Performs the computation for this variable
+		this.doRenderTarget( variable.material, variable.renderTargets[ nextTextureIndex ] );
 
-			}
+		variable.currentTextureIndex = nextTextureIndex;
+
+	};
+
+	this.compute = function () {
 
-			// Performs the computation for this variable
-			this.doRenderTarget( variable.material, variable.renderTargets[ nextTextureIndex ] );
+		for ( var i = 0, il = this.variables.length; i < il; i ++ ) {
+
+			this.computeVariable( this.variables[ i ] );
 
 		}
 
-		this.currentTextureIndex = nextTextureIndex;
 	};
 
-	this.getCurrentRenderTarget = function( variable ) {
+	this.getCurrentRenderTarget = function ( variable ) {
 
-		return variable.renderTargets[ this.currentTextureIndex ];
+		return variable.renderTargets[ variable.currentTextureIndex ];
 
 	};
 
-	this.getAlternateRenderTarget = function( variable ) {
+	this.getAlternateRenderTarget = function ( variable ) {
 
-		return variable.renderTargets[ this.currentTextureIndex === 0 ? 1 : 0 ];
+		return variable.renderTargets[ variable.currentTextureIndex === 0 ? 1 : 0 ];
 
 	};
 
-	function addResolutionDefine( materialShader ) {
+	function getResolutionDefineString( w, h ) {
 
-		materialShader.defines.resolution = 'vec2( ' + sizeX.toFixed( 1 ) + ', ' + sizeY.toFixed( 1 ) + " )";
+		return "\n#define resolution vec2(" + w.toFixed( 1 ) + ", " + h.toFixed( 1 ) + ")\n";
 
 	}
-	this.addResolutionDefine = addResolutionDefine;
 
+	this.getResolutionDefineString = getResolutionDefineString;
 
 	// The following functions can be used to compute things manually
 
-	function createShaderMaterial( computeFragmentShader, uniforms ) {
+	function createShaderMaterial( computeFragmentShader, uniforms, sizeVariableX, sizeVariableY ) {
 
 		uniforms = uniforms || {};
 
+		// set the proper resolution for the fragment shader
+		sizeVariableX = sizeVariableX || sizeX;
+		sizeVariableY = sizeVariableY || sizeY;
+
+		// add the resolution parameter
+		computeFragmentShader = getResolutionDefineString( sizeVariableX, sizeVariableY ) + computeFragmentShader;
+
 		var material = new THREE.ShaderMaterial( {
 			uniforms: uniforms,
 			vertexShader: getPassThroughVertexShader(),
 			fragmentShader: computeFragmentShader
 		} );
 
-		addResolutionDefine( material );
-
 		return material;
+
 	}
+
 	this.createShaderMaterial = createShaderMaterial;
 
-	this.createRenderTarget = function( sizeXTexture, sizeYTexture, wrapS, wrapT, minFilter, magFilter ) {
+	this.createRenderTarget = function ( sizeXTexture, sizeYTexture, wrapS, wrapT, minFilter, magFilter ) {
 
 		sizeXTexture = sizeXTexture || sizeX;
 		sizeYTexture = sizeYTexture || sizeY;
 
 		wrapS = wrapS || THREE.ClampToEdgeWrapping;
 		wrapT = wrapT || THREE.ClampToEdgeWrapping;
 
-		minFilter = minFilter || THREE.NearestFilter;
-		magFilter = magFilter || THREE.NearestFilter;
+		minFilter = minFilter || this.magFilter;
```

Comment:
Typo. `this.minFilter`. I am not sure it is advisable to support filters other than `THREE.NearestFilter`. Do you have a use case?